packages/cli/src/commands/setup/ui/libraries/windicss.js

```diff
@@ -0,0 +1,147 @@
+
+  const packages = ['windicss-webpack-plugin', 'windicss']
+
+  const tasks = new Listr([
+    {
+      title: 'Installing packages...',
+      skip: () => !install,
+      task: () => {
+        return new Listr([
+          {
+            title: `Install ${packages.join(', ')}`,
+            task: async () => {
+              await execa('yarn', [
+                'workspace',
+                'web',
+                'add',
+                '-D',
+                ...packages,
+              ])
+            },
+          },
+        ])
+      },
+    },
+    {
+      title: 'Setup Webpack...',
+      task: () => {
+        return new Listr([
+          {
+            title: 'Setup Webpack',
+            task: async () => {
+              await execa('yarn', [
+                'redwood',
+                'setup',
+                'webpack'
+              ])
+            },
+          },
+          {
+            title: 'Configure WindiCSS',
+            task: async () => {
+              const webpackConfigPath = path.join(
+                rwPaths.web.config,
+                'webpack.config.js'
+              )
+              const webpackConfig = fs.readFileSync(webpackConfigPath, 'utf-8')
+              const newWebpackConfig = `const WindiCSSWebpackPlugin = require('windicss-webpack-plugin')\n`
+                + webpackConfig.replace(
+                  '// config.plugins.push(YOUR_PLUGIN)',
+                  '// config.plugins.push(YOUR_PLUGIN)\n  config.plugins.push(new WindiCSSWebpackPlugin())'
+                )
+              fs.writeFileSync(webpackConfigPath, newWebpackConfig)
+            },
+          },
+        ])
+      },
+    },
+    {
+      title: 'Initializing WindiCSS...',
+      task: async () => {
+        const windiConfigPath = path.join(
+          rwPaths.web.config,
+          'windi.config.js'
+        )
+
+        if (fs.existsSync(windiConfigPath)) {
+          if (force) {
+            fs.unlinkSync(windiConfigPath)
+          } else {
+            throw new Error(
+              'Windicss config already exists.\nUse --force to override existing config.'
+            )
+          }
+        }
+
+        const windiConfig = `
+          import { defineConfig } from 'windicss/helpers'
+
+          export default defineConfig({
+            extract: {
+              include: ['**/*.{jsx,tsx,css}'],
+              exclude: ['node_modules', '.git'],
+            },
+          })
+        `.split('\n').map((line) => line.trim()).join('\n')
+        fs.writeFileSync(windiConfigPath, windiConfig)
+      },
+    },
+    {
+      title: 'Adding import to App.tsx...',
+      task: (_ctx, task) => {
+        const APP_TSX_PATH = path.join(rwPaths.web.src, 'App.tsx')
+        const appTsx = fs.readFileSync(APP_TSX_PATH, 'utf-8')
+
+        if (windiImportsExist(appTsx)) {
+          task.skip('Imports already exist in App.tsx')
+        } else {
+          const newAppTsx = `import 'windi.css'\n` + appTsx
```

Comment:
This would put the windi import at the top of the file:

```diff
+ import 'windi.css'
  import { FatalErrorBoundary, RedwoodProvider } from '@redwoodjs/web'
  import { RedwoodApolloProvider } from '@redwoodjs/web/apollo'
```

It'd definitely be harder, but maybe we could put the the import before the App declaration, near where you normally see `import './index.css'`.

Reply:
Sure! I was just thinking about something: what if the developer removed the "index.css" import, or even for the webpack part, if he removed the comments? I just thought about it now, but it can be problematic.

Reply:
I think removing `index.css` could be problematic like you said, especially if someone's setting it up later and moving styles over. See my suggestion here: https://github.com/redwoodjs/redwood/pull/5194#discussion_r851709313.